### PR TITLE
CEP-1261

### PR DESCRIPTION
**Pull request by dilini-muthumala**

### CODE REVIEW

File: components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/server/TCPEventServer.java [hunk 1]
```diff
@@ -42,12 +42,14 @@
     private TCPEventServerConfig tcpEventServerConfig = new TCPEventServerConfig(7211);
     private ExecutorService executorService;
     private StreamCallback streamCallback;
+    private ConnectionCallback connectionCallback;
     private ServerWorker serverWorker;
     private Map<String, StreamRuntimeInfo> streamRuntimeInfoMap = new ConcurrentHashMap<String, StreamRuntimeInfo>();
 
-    public TCPEventServer(TCPEventServerConfig tcpEventServerConfig, StreamCallback streamCallback) {
-        this.tcpEventServerConfig = tcpEventServerConfig;
+    public TCPEventServer(TCPEventServerConfig TCPEventServerConfig, StreamCallback streamCallback, ConnectionCallback connectionCallback) {
```
Comment (suhothayan) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/server/TCPEventServer.java]: TCPEventServerConfig should be tcpEventServerConfig

File: components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java [hunk 2]
```diff
@@ -277,6 +292,11 @@ private void disconnect() {
             }
         } catch (IOException e) {
             log.debug("Error while closing socket to " + hostUrl + " : " + e.getMessage(), e);
+        }  finally {
+            if(connectionCallback != null){
+                connectionCallback.onCepReceiverDisconnect();
+                log.info("--------------------------TCPEventPublisher onclose invoked.");//todo: delete later
```
Comment (suhothayan) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java]: Don't forget to delete this

Comment (dilini-muthumala) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java]: It's deleted after testing. Thanks for pointing out.

File: components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/server/TCPEventServer.java (hunk 1 above)
Comment (dilini-muthumala) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/server/TCPEventServer.java]: Fixed. Thanks for pointing out.

File: components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java (hunk 2 above)
Comment (suhothayan) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java]: this is not yet deleted

Comment (dilini-muthumala) [inline components/event-processor-manager/org.wso2.carbon.event.processor.manager.commons/src/main/java/org/wso2/carbon/event/processor/manager/commons/transport/client/TCPEventPublisher.java]: oops! I searched & deleted all the comments but somehow have missed this.. :(
Deleted now!

**Pull request merged**